massa-models/src/serialization.rs

```diff
@@ -49,31 +50,28 @@ impl SerializeMinBEInt for u64 {
 
 /// Deserialize min big endian
 pub trait DeserializeMinBEInt: Sized {
-    /// min big endian integer base size
-    const MIN_BE_INT_BASE_SIZE: usize;
-
-    /// Compute the minimal big endian deserialization size
-    fn be_bytes_min_length(max_value: Self) -> usize;
-
     /// Deserializes a minimally sized big endian integer to Self from the provided buffer and checks that its value is within given bounds.
     /// In case of success, return the deserialized data and the number of bytes read
     fn from_be_bytes_min(buffer: &[u8], max_value: Self) -> Result<(Self, usize), ModelsError>;
 }
 
-impl DeserializeMinBEInt for u32 {
-    const MIN_BE_INT_BASE_SIZE: usize = 4;
-
-    fn be_bytes_min_length(max_value: Self) -> usize {
-        Self::MIN_BE_INT_BASE_SIZE - (max_value.leading_zeros() as usize) / 8
-    }
+/// Compute the minimal big endian deserialization size
+pub const fn u32_be_bytes_min_length(max_value: u32) -> usize {
+    size_of::<u32>() - (max_value.leading_zeros() as usize) / 8
+}
 
+/// Compute the minimal big endian deserialization size
+pub const fn u64_be_bytes_min_length(max_value: u64) -> usize {
+    size_of::<u64>() - (max_value.leading_zeros() as usize) / 8
+}
+impl DeserializeMinBEInt for u32 {
     fn from_be_bytes_min(buffer: &[u8], max_value: Self) -> Result<(Self, usize), ModelsError> {
-        let read_bytes = Self::be_bytes_min_length(max_value);
-        let skip_bytes = Self::MIN_BE_INT_BASE_SIZE - read_bytes;
+        let read_bytes = u32_be_bytes_min_length(max_value);
```

Comment:
const methods as part of traits is... difficult/impossible. Tried a macro, but to make it work "elegantly" would be a poor use of time. Given it's just two types, and not something that is likely to be generalised in the (near) future (from what I can tell), putting in a todo/follow-up is likely to be a distraction, though happy to be corrected.